Foundation/NSURLSession/http/HTTPURLProtocol.swift

```diff
@@ -439,8 +439,14 @@ extension _HTTPURLProtocol {
 extension _HTTPURLProtocol: _EasyHandleDelegate {
     
     func didReceive(data: Data) -> _EasyHandle._Action {
-        guard case .transferInProgress(let ts) = internalState else { fatalError("Received body data, but no transfer in progress.") }
-        guard ts.isHeaderComplete else { fatalError("Received body data, but the header is not complete, yet.") }
+        guard case .transferInProgress(var ts) = internalState else { fatalError("Received body data, but no transfer in progress.") }
+        if !ts.isHeaderComplete {
+            ts.response = HTTPURLResponse(url: ts.url, statusCode: 200, httpVersion: "0.9", headerFields: [:])
```

Comment:
Is this valid in the three different scenarios that you have pointed out?
1. When the server sends a HTTP/0.9 response aka "only the body"
2. When the server is running HTTP/1.1 but sends a malformed response
3. When the server running HTTP/2 sends a response to client HTTP/1.1 
... I'm specifically concerned about  the `httpVersion: "0.9"`